### PR TITLE
Gr3.1

### PR DESCRIPTION
**Pull request by david-dearden-ssc**
# AWS CAC Solution PR Template

## Pull Request Type
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation Update
- [ ] Other (Please specify)

## Description
Implementation of Guard Rails:

* 3.1: Confirm that administrative access to cloud environments is from approved and trusted locations and devices

Cleanup and Refactoring:

* Remove duplicate entry in `arch/templates/ConformancePack.yaml` for `GC08CheckCloudDeploymentGuideDocumentConfigRule`
* Update logging in the audit report lambda (Set the log level and log the event)
* Update variable names in the audit report lambda to separate words with `_` (and fix spelling mistakes)
* Update the event log in the create role lambda to be consistent with the other lambdas
* Updated the annotation in the gc13 emergency account alerts to include more info when the s3 object does not exist
* Update `list_cloud_trails` in the gc01 check monitoring and logging lambda to use the `next_token` and to be setup in the loop style

## Related Issue(s)
List any related issues here. Include `Fixes #issue_number` or `Closes #issue_number` if applicable.

## How Has This Been Tested?
Describe how you have tested these changes. Include details of the testing environment and the tests you ran to verify the functionality.

### CloudFormation Linter Results
Include the output of the CloudFormation linter (if used).

### Deployment Test Results
Provide details of any deployment tests you have conducted.

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have updated any related documentation.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My changes generate no new warnings.
- [ ] I have added comments to my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes require a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have checked that the final deliverable aligns with the CloudFormation best practices.

## Additional Notes
Include any additional information that you think is important for reviewers.

### CODE REVIEW

File: arch/templates/ConformancePack.yaml [hunk 1]
```diff
@@ -791,42 +840,6 @@ Resources:
         SourceDetails:
           - EventSource: "aws.config"
             MessageType: "ScheduledNotification"
-  GC08CheckCloudDeploymentGuideDocumentConfigRule:
```
Comment (david-dearden-ssc) [inline arch/templates/ConformancePack.yaml:794]: Deleted because it is a duplicate of the entry above.

File: src/lambda/aws_compile_audit_report/app.py
Comment (david-dearden-ssc) [inline src/lambda/aws_compile_audit_report/app.py:1]: * Set log level
* Log the event
* Cleanup spelling mistakes
* Update variable names to separate words with `_`

File: src/lambda/aws_create_role/app.py [hunk 2]
```diff
@@ -212,7 +212,7 @@ def lambda_handler(event, context):
     context -- the context variable given in the lambda handler
     """
     response_data = {}
-    logger.info('Event: {}'.format(event))
+    logger.info("Received Event: %s", json.dumps(event, indent=2))
```
Comment (david-dearden-ssc) [inline src/lambda/aws_create_role/app.py:215]: Updating the event log to be the same as the other lambdas.

File: src/lambda/gc01_check_monitoring_and_logging/app.py [hunk 3]
```diff
@@ -74,26 +74,26 @@ def evaluate_parameters(rule_parameters):
...
-            response = AWS_CLOUDTRAIL_CLIENT.list_trails()
-            trails = trails + response.get("Trails")
+        trails = []
+        next_token = None
+        while True:
+            response = AWS_CLOUDTRAIL_CLIENT.list_trails() if not next_token else AWS_CLOUDTRAIL_CLIENT.list_trails(NextToken=next_token)
+            trails = trails + response.get("Trails", [])
             next_token = response.get("NextToken")
+            if not next_token:
+                break
```
Comment (david-dearden-ssc) [inline src/lambda/gc01_check_monitoring_and_logging/app.py:87]: Updated to use the loop style, but also needed to pass in the next token to the `list_trails` call when it is available.

File: src/lambda/gc13_emergency_account_alerts/app.py [hunk 4]
```diff
@@ -311,16 +311,9 @@ def lambda_handler(event, context):
...
-        evaluations.append(
-            build_evaluation(
-                event["accountId"],
-                "NON_COMPLIANT",
-                event,
-                resource_type=DEFAULT_RESOURCE_TYPE,
-                annotation=f"No {file_param_name} input provided.",
-            )
-        )
+        annotation = f"No file found for s3 path '{rule_names_file_path}' via '{file_param_name}' input parameter."
```
Comment (david-dearden-ssc) [inline src/lambda/gc13_emergency_account_alerts/app.py:314]: Update annotation to show more information about the required input.

**Review by TylerTMizuyabu — APPROVED** [review]

**Review by TylerTMizuyabu — APPROVED** [review]

**Review by dan-mccrady — APPROVED** [review]

**Pull request merged**